pkg/webhooks/prometheusrule/prometheusrule.go

```diff
@@ -0,0 +1,205 @@
+	"github.com/openshift/managed-cluster-validating-webhooks/pkg/webhooks/utils"
+	admissionv1 "k8s.io/api/admission/v1"
+	admissionregv1 "k8s.io/api/admissionregistration/v1"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	admissionctl "sigs.k8s.io/controller-runtime/pkg/webhook/admission"
+
+	logf "sigs.k8s.io/controller-runtime/pkg/log"
+)
+
+const (
+	WebhookName               string = "promrule-validation"
+	docString                 string = `Managed OpenShift Customers may not create promrule in namespaces managed by SRE .`
+	privilegedServiceAccounts string = `^system:serviceaccounts:(kube.*|openshift.*|default|redhat.*|osde2e-[a-z0-9]{5})`
+	managednamespace          string = `openshift-*|kube-*|redhat-*`
+)
+
+var (
+	timeout int32 = 2
+	//log                   = logf.Log.WithName(WebhookName)
+	clusterAdminUsers = []string{"kube:admin", "system:admin", "backplane-cluster-admin"}
+	sreAdminGroups    = []string{"system:serviceaccounts:openshift-backplane-srep"}
+	//privilegedServiceAccountsRe = regexp.MustCompile(privilegedServiceAccounts)
+
+	scope = admissionregv1.ClusterScope
+	rules = []admissionregv1.RuleWithOperations{
+		{
+			Operations: []admissionregv1.OperationType{"CREATE"},
+			Rule: admissionregv1.Rule{
+				APIGroups:   []string{"monitoring.coreos.com"},
+				APIVersions: []string{"*"},
+				Resources:   []string{"prometheusrule"},
+				Scope:       &scope,
+			},
+		},
+		{
+			Operations: []admissionregv1.OperationType{"UPDATE"},
+			Rule: admissionregv1.Rule{
+				APIGroups:   []string{"monitoring.coreos.com"},
+				APIVersions: []string{"*"},
+				Resources:   []string{"prometheusrule"},
+				Scope:       &scope,
+			},
+		},
+		{
+			Operations: []admissionregv1.OperationType{"DELETE"},
+			Rule: admissionregv1.Rule{
+				APIGroups:   []string{"monitoring.coreos.com"},
+				APIVersions: []string{"*"},
+				Resources:   []string{"prometheusrule"},
+				Scope:       &scope,
+			},
+		},
+	}
+	log = logf.Log.WithName(WebhookName)
+)
+
+// PrometheusruleWebhook validates a promrulechange
+type PrometheusruleWebhook struct {
+	s runtime.Scheme
+}
+
+// NewWebhook creates the new webhook
+func NewWebhook() *PrometheusruleWebhook {
+	scheme := runtime.NewScheme()
+	return &PrometheusruleWebhook{
+		s: *scheme,
+	}
+}
+
+// Authorized implements Webhook interface
+func (s *PrometheusruleWebhook) Authorized(request admissionctl.Request) admissionctl.Response {
+	return s.authorized(request)
+}
+
+func (s *PrometheusruleWebhook) authorized(request admissionctl.Request) admissionctl.Response {
+	var ret admissionctl.Response
+
+	ns, err := s.rendermanagednamespace(request)
+	if err != nil {
+		log.Error(err, "Couldn't render a Namespace from the incoming request")
+		return admissionctl.Errored(http.StatusBadRequest, err)
+	}
+
+	if hookconfig.IsPrivilegedNamespace(ns.GetName()) && !isclusterAdminUsers(request) {
+		switch request.Operation {
```

Comment:
Do we need this switch case? We block on all operations so this seems unnecessary?